internal/ctrlutils/ingress-status.go

```diff
@@ -44,32 +44,27 @@ func PullConfigUpdate(
 ) {
 	ips, hostname, err := RunningAddresses(ctx, kubeConfig, publishService, publishAddresses)
 	if err != nil {
-		log.Error(err, "failed to determine kong proxy external ips/hostnames.")
-		return
+		panic(err)
```

Comment:
We shouldn't panic here, we should change the signature of the `PullConfigUpdate` function to include an `error` in the return values.

The good news is this is all `internal/` so we have a pretty free room to rework this however we like.

Reply:
1ef989f7fb18916a06d2b0864ac5964d42610edf